device/esp_tinyusb/include_private/descriptors_control.h

```diff
@@ -6,23 +6,46 @@
 
 #pragma once
 
-#include "tusb.h"
+// #include "tusb.h"
+#include "tinyusb.h"
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+#define USB_STRING_DESCRIPTOR_ARRAY_SIZE            8 // Max 8 string descriptors for a device. LANGID, Manufacturer, Product, Serial number + 4 user defined
+
+/**
+ * @brief USB Device descriptor pointers for tinyusb configuration
+ *
+ */
+typedef struct {
+    const tusb_desc_device_t *dev;      /*!< Pointer to device descriptor */
+    union {
+        const uint8_t *cfg;             /*!< Pointer to FS configuration descriptor when device one-speed only */
+        const uint8_t *hs_cfg;          /*!< Pointer to HS configuration descriptor when device one-speed only */
+    };
+#if (TUD_OPT_HIGH_SPEED)
+    const uint8_t *fs_cfg;              /*!< Pointer to FS configuration descriptor when device support HS */
+    tusb_desc_device_qualifier_t *qualifier;            /*!< Pointer to Qualifier descriptor */
+    bool qualifier_alloc_internally;    /*!< Qualifier descriptor was allocated  internally */
```

Comment:
Because if it is not provided by user, it assembling just right there (Not to keep both descriptors and use the memory in case is has been presented by user). With copying data from Device Descriptor (that is why it is not `const`)
But in case changing the way of how descriptors is kept (static or dynamic) this part also could be changed and can be done simpler.